docs/proposals/PROPOSAL_B.md

```diff
@@ -0,0 +1,287 @@
+
+### Definitions
+
+* image-compatibility artifact - an OCI artifact that stores the compatibility spec.
+* Compatibility Spec - a json object that represents a graph of namespaces.
+* Namespace - a validation area under which users can describe plugin specs that are consumed by plugins (executable binaries) as input.
+* Plugin - an executable binary that extracts data and checks compatibility on a host.
+
+A new imageCompatibility artifact type that contains a layer with a JSON compatibility object.
+The object represents a graph of namespaces.
+Namespace describes the validation area and is another object containing data that provides plugins specification for the corresponding plugins.
+A plugin is an executable binary that validates the host and returns information about success or failure.
+
+### Compatibility Spec
+
+Compatibility spec is a graph that can have multiple roots with different paths.
+If one namespace in the path fails, the path is incompatible with the host.
+Namespace fails if at least one plugin reports an error.
+If all paths fail, the host is not compatible with the container.
+
+The host is compatible with the container if at least one path is completed successfully.
+
+#### Schema
+
+- **`schemaVersion`** *int*
+
+  This REQUIRED property specifies the image manifest schema version. For this version of the specification.
+
+- **`namespaces`** *string-object map*
+
+  This REQUIRED property specifies a map of validation areas.
+
+  - **`<namespace-name>`** *object*
+
+    This REQUIRED property specifies the validation area.
+
+    - **`plugins`** *string-object*
+
+      This OPTIONAL property is a list of plugins specifications.
+
+        - **`<plugin-name>`** *object*
+
+          This REQUIRED property specifies the plugin specification for plugin executable binary that validates the host and returns information about success or failure
+
+          - **`source`** *string*
+
+            This OPTIONAL property specifies an HTTP address of the plugin.
+
+          - **`data`** *object*
+
+            This OPTIONAL property specifies the input data for plugin executable binary. 
+
+    - **`refs`** *object*
+
+      This OPTIONAL property specifies the references to other namespaces.
+
+      - **`all`** *array of strings*
+
+        This OPTIONAL property specifies a list of all required namespaces that have to pass that container is considered to be compatible with a host.
+
+      - **`oneOf`** *array of strings*
+
+        This OPTIONAL property specifies a list of namespaces with a condition that at least one namespace has to pass that container is considered to be compatible with a host.
+
+      - **`noneOf`** *array of strings* 
+
+        This OPTIONAL property specifies a list of namespaces that cannot pass that container is considered to be compatible with a host.
+
+##### Example
+
+<details>
+  <summary>GPU pass-through</summary>
+
+```json
```

Comment:
What is refs saying here? That if I run the kernelVersion plugin, I _must_ run (and have pass) those others as well?

Reply:
If you use refs `all`, you are right. There are also two additional ones: "oneOf", "noneOf", which describe the passing condition.